src/shared/utils/helpers/get-query-params.ts

```diff
@@ -1,21 +1,28 @@
-import { isBrowser } from "@utils/browser";
+type Empty = NonNullable<unknown>;
+
+type QueryMapping<PropsT, FallbacksT extends Empty> = {
+  [K in keyof PropsT]-?: (
```

Comment:
Why the `-` before the `?`? I've never seen that syntax used in typescript before.

Reply:
Makes specifying a mapping for each property manadatory. Works like `Required<T>`:

```typescript
/**
 * Make all properties in T required
 */
type Required<T> = {
    [P in keyof T]-?: T[P];
};
```